tests/scripts/all.sh

```diff
@@ -1052,9 +1053,7 @@ component_test_m32_o1 () {
     # Build again with -O1, to compile in the i386 specific inline assembly
     msg "build: i386, make, gcc -O1 (ASan build)" # ~ 30s
     scripts/config.pl full
-    scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE
-    scripts/config.pl unset MBEDTLS_MEMORY_BUFFER_ALLOC_C
-    scripts/config.pl unset MBEDTLS_MEMORY_DEBUG
```

Comment:
Did we intentionally remove `unset MBEDTLS_MEMORY_BACKTRACE` and `unset MBEDTLS_MEMORY_DEBUG`? I don't see any reasoning for why we should and do remove these options in the commit message.

Reply:
These options have no effect if `MBEDTLS_MEMORY_BUFFER_ALLOC_C` is unset.

Reply:
Yes, but that isn't in the commit message so someone browsing history may see this as an accidental or inadvertent change. It's a refactoring separate from purpose of "Disable MEMORY_BUFFER_ALLOC with ASan " and proabbly ought to be in its own commit, but at least explaining the refactor in the commit as currently composed would be a good improvement.

Reply:
Ok, I made it a separate commit.